src/servers/GatewayServer/gatewayserver.threaded.ts

```diff
@@ -0,0 +1,100 @@
+// ======================================================================
+//
+//   GNU GENERAL PUBLIC LICENSE
+//   Version 3, 29 June 2007
+//   copyright (C) 2020 - 2021 Quentin Gruber
+//   copyright (C) 2021 - 2024 H1emu community
+//
+//   https://github.com/QuentinGruber/h1z1-server
+//   https://www.npmjs.com/package/h1z1-server
+//
+//   Based on https://github.com/psemu/soe-network
+// ======================================================================
+
+import { EventEmitter } from "node:events";
+import SOEClient from "../SoeServer/soeclient";
+import { SOEOutputChannels } from "servers/SoeServer/soeoutputstream";
+import { Worker, MessageChannel } from "node:worker_threads";
+
+export enum GatewayServerThreadedInternalEvents {
+  START,
+  STOP
+}
+
+export class GatewayServerThreaded extends EventEmitter {
+  // TODO:
+  // implements GatewayServer
```

Comment:
I wanted that for type safety but the typescript compiler probably doesn't allow such advanced weird stuff